internal/wsproxy/recorder.go

```diff
@@ -21,6 +22,24 @@ var (
 	errAlreadyFinished     = errors.New("recording already finished")
 )
 
+var (
+	recordedSessionDuration prometheus.Histogram
+	initializeMetricsOnce   sync.Once
+)
+
+func RegisterRecordedSessionMetrics(namespace string, registry *prometheus.Registry) {
+	initializeMetricsOnce.Do(func() {
+		recordedSessionDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: namespace,
+			Name:      "recorded_session_duration_seconds",
+			Help:      "Duration of WebSocket session in seconds",
+			Buckets:   []float64{1, 5, 10, 30, 60, 300, 600, 1800, 3600},
+		})
+	})
```

Comment:
We detected a data race in the tests (see [CI](https://github.com/Twingate/kubernetes-access-gateway/actions/runs/16132786927/job/45523187155)), using `sync.Once` here should fix it.

I think it's fair to assume we won't initialize these metrics again with a different namespace. So this solution should be fine. I am open to other alternatives